src/store/features/tokenSlice.tsx

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable no-param-reassign */
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+interface ITokenState {
+  token: string | null;
+  isAuthenticated: boolean;
+}
+
+const initialState: ITokenState = {
+  token: null,
+  isAuthenticated: false,
+};
+
+const tokenSlice = createSlice({
+  name: "auth",
+  initialState,
+  reducers: {
+    login: (state, action: PayloadAction<string>) => {
+      state.token = action.payload;
+      state.isAuthenticated = true;
```

Comment:
What is the difference between userSlice.isLoggedIn and tokenSlice.isAuthenticated?
It should be only tokenSlice.isAuthenticated. Token should be the only source of truth here

Reply:
done